tests/phpunit/tests/user.php

```diff
@@ -915,6 +915,69 @@ public function test_wp_insert_user_should_reject_user_login_that_matches_existi
 		$this->assertSame( 'existing_user_email_as_login', $user_id->get_error_code() );
 	}
 
+	/**
+	 * @ticket 57967
+	 */
+	public function test_wp_update_user_should_allow_user_login_with_same_user_email() {
+		$user_email = str_repeat( 'a', 5 ) . '@example.com';
+
+		$user_id = wp_insert_user(
+			array(
+				'user_login'    => $user_email,
+				'user_email'    => $user_email,
+				'user_pass'     => 'whatever',
+				'user_nicename' => 'whatever',
+			)
+		);
+
+		$existing_id = wp_update_user(
+			array(
+				'ID'         => $user_id,
+				'first_name' => 'Bob',
+			)
+		);
+
+		$this->assertNotWPError( $existing_id );
+		$this->assertSame( $existing_id, $user_id );
```

Comment:
```suggestion
		$this->assertNotWPError( $existing_id, 'A WP_Error object was not returned.' );
		$this->assertSame( $existing_id, $user_id, 'The user ID to be updated and the existing user ID do not match.' );
```
Per [Using Assertions](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions)
> All PHPUnit assertions, as well as all WordPress custom assertions, allow for a `$message` parameter to be passed. This message will be displayed when the assertion fails and can help immensely when debugging a test. This parameter should always be used if more than one assertion is used in a test method.